src/main/scala/ch/epfl/lamp/CourseraBuild.scala

```diff
@@ -126,35 +123,20 @@ trait CourseraBuild extends sbt.Build {
   val grade = TaskKey[Unit]("grade")
 
   // mapR: submit the grade / feedback in any case, also on failure
-  lazy val gradeSetting = grade <<= (gradingUUID, scalaTestSubmission, styleCheckSubmission, apiKey, gradeProjectDetails, streams) mapR { (uuidR, sts, scs, apiKeyR, projectDetailsR, s) =>
-    val Value(uuid) = uuidR
+  lazy val gradeSetting = grade <<= (scalaTestSubmission, styleCheckSubmission, gradeProjectDetails, streams) mapR { (sts, scs, projectDetailsR, s) =>
     val logOpt = s match {
       case Value(v) => Some(v.log)
       case _ => None
     }
-    logOpt.foreach(_.info(GradingFeedback.feedbackString(uuid, html = false)))
+    val feedback = GradingFeedback.feedbackString(html = false)
+    // Output to logs for instructor debugging purposes.
+    logOpt.foreach(_.info(feedback))
     val Value(projectDetails) = projectDetailsR
-    apiKeyR match {
-      case Value(apiKey) if (!apiKey.isEmpty) =>
-        logOpt.foreach(_.debug("Course Id for submission: " + projectDetails.courseId))
-        logOpt.foreach(_.debug("Corresponding API key: " + apiKey))
-        // if build failed early, we did not even get the api key from the submission queue
-        if (!GradingFeedback.apiState.isEmpty && !Settings.offlineMode) {
-          val scoreString = "%.2f".format(GradingFeedback.totalScore)
-          CourseraHttp.submitGrade(GradingFeedback.feedbackString(uuid), scoreString, GradingFeedback.apiState, apiKey, projectDetails, logOpt) match {
-            case Failure(msgs) =>
-              sys.error(msgs.list.mkString("\n"))
-            case _ =>
-              ()
-          }
-        } else if (Settings.offlineMode) {
-          logOpt.foreach(_.info(" \nSettings.offlineMode enabled, not uploading the feedback"))
```

Comment:
@vjovanov I understand that `offlineMode` meant without talking to Coursera, but now it'd never talk to Coursera in the first place. I could add it back in so that it skips unzipping the jar.

Because `sbt-coursera` has now been pulled out into a separate plugin, wouldn't it make most sense that if we wanted to support it that we'd make it a settings key instead so that it could be easily modified when someone is trying to debug something?